<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>


  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find wheelchair_door)/param/door_costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find wheelchair_door)/param/door_costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find wheelchair_door)/param/door_local_costmap_params.yaml" command="load" />
    <rosparam file="$(find wheelchair_door)/param/door_global_costmap_params.yaml" command="load" />
    <rosparam file="$(find wheelchair_door)/param/door_move_base_params.yaml" command="load" />
    <!-- dwa  -->
    <rosparam file="$(find wheelchair_door)/param/door_dwa_local_planner_params.yaml" command="load" />
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
   
    
    <!-- teb 
    <rosparam file="$(find wheelchair_door)/param/door_teb_local_planner_params.yaml" command="load" />
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <rosparam file="$(find wheelchair_door)/param/door_costmap_converter_params.yaml" command="load" />
    -->
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    
  </node>
</launch>
